node/cmd/ccq/query_server.go

```diff
@@ -281,7 +281,12 @@ func runQueryServer(cmd *cobra.Command, args []string) {
 
 	// Start watching for permissions file updates.
 	errC := make(chan error)
-	permissions.StartWatcher(ctx, logger, errC)
+	permWatcherErr := permissions.StartWatcher(ctx, logger, errC)
+	if permWatcherErr != nil {
+		// Cleanup p2p connection.
+		cancel()
+		logger.Fatal("Could not start permissions file watcher", zap.Error(err))
```

Comment:
Is `Fatal` intentional and correct? Should it be `Error` instead?

Reply:
Yeah I think it could be Error + exit rather than Fatal.

Reply:
Actually I think Fatal is OK here:
- Fatal does log + os.Exit() under the hood
- Fatal is used a ton of other times in this file